topic-operator/src/main/java/io/strimzi/operator/topic/KafkaImpl.java

```diff
@@ -222,75 +56,88 @@ protected void queueWork(Work work) {
         Promise<Void> handler = Promise.promise();
         LOGGER.debug("Deleting topic {}", topicName);
         KafkaFuture<Void> future = adminClient.deleteTopics(
-                Collections.singleton(topicName.toString())).values().get(topicName.toString());
-        queueWork(new UniWork<>("deleteTopic", future, handler));
-        return handler.future().compose(ig ->
+                singleton(topicName.toString())).values().get(topicName.toString());
+        mapFuture(future).compose(ig ->
                 Util.waitFor(vertx, "deleted sync " + topicName, "deleted", 1000, 120_000, () -> {
                     try {
-                        return adminClient.describeTopics(Collections.singleton(topicName.toString())).all().get().get(topicName.toString()) == null;
+                        return adminClient.describeTopics(singleton(topicName.toString())).all().get().get(topicName.toString()) == null;
```

Comment:
it's something not actually related to the fix but tbh I am starting to really hate the "static import". From the code, it's not clear that `singleton` is coming from `Collections` but it could be a private method doing something related to singleton pattern. Tbh I would prefer to be explicit. Of course, it's not a change you have to do if you like "static import". I think that it makes the code less readable.

Reply:
We've discussed this before, and I think it depends on the familiarity of the methods being imported. If it was `emptyList()` or `asList()` I'm guessing you'd be a little more comfortable with it, because you'd be more prepared to guess it was `Collections.emptyList()` or `Arrays.asList()`, because those methods are more familiar to you. Personally I know those APIs pretty well, so I guess I'm more comfortable with code like this, and I don't especially like seeing those methods class qualified. 

If you feel strongly about it maybe we should formalise which classes & methods we consider OK for static imports, and use https://checkstyle.sourceforge.io/config_imports.html#AvoidStaticImport to prevent static imports of anything else. At least that would put everyone on the same page and make the code base consistent wrt static imports.